### ISSUE TITLE
Slow deployments

### ISSUE THREAD

**Issue by sman591** [codeRIT/brickhack.io#62]
Our deployments from Travis to OpenShift are taking anywhere from 5 to 15 minutes, most of which results in downtime in production.

Currently, gems are being pre-packaged by Travis before it deploys to the server, which saves some time by not requiring OpenShift to download all of the gems. However, the slowest process is pre-compiling assets. It was attempted earlier today to make Travis handle this process, but it did not succeed (https://github.com/codeRIT/brickhack.io/commit/9214164d4a67514d06960ec2639b90b833cc7164)

**Comment by sman591** [comment]
Slowly getting there. Locally, this works when committed to master.

```bash
export RAILS_ENV=production
bundle install --deployment --without 'development test'
bundle cache
bundle exec rake assets:precompile:all RAILS_GROUPS=assets
git add -f bundle/cache/
git add -f public/assets/
git commit -m "Prepared for deployment"
```

**Comment by sman591** [comment]
Gems are now cached in the repo as of fd58c92dbb0dd67e5121d0c0444ea11de28344ff and were activated in travis as of 4fef7c5a40f0c15f96fe79fde4c3bd4c3fdb2ec0. Deploying to production, this saves ~3 minutes on total build + deploy time.

**Comment by sman591** [comment]
Resolved in #260. Deployments now take around 9-13 minutes for Travis to test & deploy. All with **zero** downtime.
